src/Form/DataTransformer/ModelToIdPropertyTransformer.php

```diff
@@ -110,13 +107,23 @@ public function reverseTransform($value)
 
         foreach ($value as $key => $id) {
             if ('_labels' === $key) {
+                unset($value[$key]);
+
                 continue;
             }
 
-            $collection->add($this->modelManager->find($this->className, $id));
+            $value[$key] = (string) $id;
+        }
+
+        $query = $this->modelManager->createQuery($this->className);
+        $this->modelManager->addIdentifiersToQuery($this->className, $query, $value);
+        $result = $this->modelManager->executeQuery($query);
+
+        if ($result instanceof \Iterator) {
```

Comment:
```suggestion
        if ($result instanceof \Traversable) {
```
I guess this would work for any other case, also for `Collection`.